### ISSUE TITLE
Alpine Linux based GraalVM Docker image

### ISSUE THREAD

**Issue by fatih-erdem** [oracle/graal#1898]
Hey, I want to use GraalVM but not the native image, just normal JIT mode.

I was searching for a Docker image for GraalVM and I saw the official one. https://hub.docker.com/r/oracle/graalvm-ce/tags

The problem is Docker image's size is about 450 MB. But for OpenJDK, I could find a Docker image based on Alpine Linux and size is around 50 MB.

Is there any plan to create the GraalVM Docker image based on Alpine Linux instead of Oracle Linux?

**Comment by fatih-erdem** [comment]
So, I can assume that there is no plan for this. 😄

**Comment by gilles-duboscq** [comment]
We'd first need to ensure we can run the JDK on alpine (JDK 8 and JDK 11).
We need to figure out the necessary patches for 8 and 11 to run on musl-based distributions (from portola / icedtea / zulu probably) and make separate builds.

**Comment by jwulf** [comment]
Spoiler alert: You can.

https://github.com/renanpalmeira/docker-graalvm-alpine

**Comment by aboullaite** [comment]
@jwulf Basically that's not running JDK on alpine/musl. he's adding `glibc` alpine package! porting JVM to `musl` is still a WIP

**Comment by petermz** [comment]
Liberica JDK runs on alpine/musl. They have 8 and 11 among others:
https://bell-sw.com/pages/downloads/?os=Linux&bitness=64-bit&architecture=x86-musl&package=jdk

**Comment by sify21** [comment]
If this is done, does it mean that we can use this directly to build a [static native image](https://www.graalvm.org/reference-manual/native-image/StaticImages/)?

**Comment by petermz** [comment]
I guess you can build musl static images even now, by cross compiling on Ubuntu, something like
`native-image --static --libc=musl`

**Comment by sify21** [comment]
Yeah I know that method, but I'm having trouble with libstdc++.a and am trying to build gcc with musl-gcc. It would be great if I can directly use this image to build a static.

**Comment by ohbus** [comment]
Looking forward to this 🤯

**Comment by tmuthuk** [comment]
Does this mean I can't install Graal VM on an a vanilla Alpine Docker image?

**Comment by fniephaus** [comment]
You may want to check out @shaunsmith's Tiny Java Containers demo and talk:
https://github.com/graalvm/graalvm-demos/tree/master/tiny-java-containers
